ProcessLib/LIE/HydroMechanics/CreateHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,305 @@
+
+#include "ProcessLib/Parameter/ConstantParameter.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+#include "ProcessLib/Utils/ProcessUtils.h"  // required for findParameter
+
+#include "HydroMechanicsProcess.h"
+#include "HydroMechanicsProcessData.h"
+
+namespace ProcessLib
+{
+namespace LIE
+{
+namespace HydroMechanics
+{
+template <unsigned GlobalDim>
+class HydroMechanicsProcess;
+
+extern template class HydroMechanicsProcess<2>;
+
+template <unsigned GlobalDim>
+std::unique_ptr<Process> createHydroMechanicsProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{process__type}
+    config.checkConfigParameter("type", "HYDRO_MECHANICS_WITH_LIE");
+    DBUG("Create HydroMechanicsProcess with LIE.");
+
+    // Process variables
+    auto const pv_conf = config.getConfigSubtree("process_variables");
+    //! \ogs_file_param_special{process__HYDRO_MECHANICS_WITH_LIE_process_variables__process_variable
+    auto range = pv_conf.getConfigParameterList<std::string>("process_variable");
+    std::vector<std::reference_wrapper<ProcessVariable>> process_variables;
+    for (std::string const& pv_name : range)
+    {
+        if (pv_name != "pressure"
+            && pv_name != "displacement"
+            && pv_name.find("displacement_jump")==std::string::npos)
+            OGS_FATAL("Found a process variable name '%s'. It should be 'displacement' or 'displacement_jumpN' or 'pressure'");
+        auto variable = std::find_if(
+            variables.cbegin(), variables.cend(),
+            [&pv_name](ProcessVariable const& v) { return v.getName() == pv_name; });
+
+        if (variable == variables.end())
+        {
+            OGS_FATAL(
+                "Could not find process variable '%s' in the provided variables "
+                "list for config tag <%s>.",
+                pv_name.c_str(), "process_variable");
+        }
+        DBUG("Found process variable \'%s\' for config tag <%s>.",
+             variable->getName().c_str(), "process_variable");
+
+        if (pv_name.find("displacement") != std::string::npos
+            && variable->getNumberOfComponents() != GlobalDim)
+        {
+            OGS_FATAL(
+                "Number of components of the process variable '%s' is different "
+                "from the displacement dimension: got %d, expected %d",
+                variable->getName().c_str(),
+                variable->getNumberOfComponents(),
+                GlobalDim);
+        }
+
+        process_variables.emplace_back(const_cast<ProcessVariable&>(*variable));
+    }
+
+    if (process_variables.size() > 3)
+        OGS_FATAL("Currently only one displacement jump is supported");
+
+    // Constitutive relation.
```

Comment:
yes it might be better to share some parts with HM process, if someone can maintain the LIE stuff. this is something you guys can discuss later because common parts in other processes should also be extracted.